rest-service/manager_rest/manager_elasticsearch.py

```diff
@@ -104,12 +103,15 @@ def _build_wildcard_condition(k, v):
                                               "ignore_unmapped": True}}, sort)
 
         if pagination:
+            pagination_size = pagination.get('size',
+                                             config.instance().max_results)
+            ManagerElasticsearch._validate_pagination(pagination_size)
             if not skip_size:
-                body['size'] = pagination.get('size', DEFAULT_SEARCH_SIZE)
+                body['size'] = pagination_size
             if 'offset' in pagination:
                 body['from'] = pagination['offset']
         elif not skip_size:
-            body['size'] = DEFAULT_SEARCH_SIZE
```

Comment:
if user provided size we'll be in "if pagination" scope. we'll get here if user didn't provided it..